File: src/transformers/features/json/JsonStringifyTransformer.ts
import { JsonStringifyProgrammer } from "../../../programmers/json/JsonStringifyProgrammer";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonStringifyTransformer {
    export const transform = GenericTransformer.scalar("stringify")(
        (project) => (modulo) => JsonStringifyProgrammer.write(project)(modulo),
    );
}

File: packages/transform/src/features/json/JsonStringifyTransformer.go
package json

import (
	corejson "github.com/samchon/typia/packages/core/src/programmers/json"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var JsonStringifyTransformer = jsonStringifyTransformerNamespace{}

type jsonStringifyTransformerNamespace struct{}

func (jsonStringifyTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corejson.JsonStringifyProgrammer.Write(call.TypeText)
}
